sensing/src/main/java/com/google/android/sensing/db/impl/DatabaseImpl.kt

```diff
@@ -67,7 +67,7 @@ internal class DatabaseImpl(context: Context, databaseConfig: DatabaseConfig) :
   }
 
   override suspend fun updateUploadRequest(uploadRequest: UploadRequest) {
-    uploadRequestDao.updateUploadRequest(uploadRequest)
+    uploadRequestDao.insertUploadRequest(uploadRequest)
```

Comment:
Just wondering why the record needs to re-inserted on upload? Also, I was found that the `id` is set to `0` during `update` transaction here. Can you please check this?

Reply:
Record is being updated using the same insert API with onConflict strategy being repace.

Internally it checks for primary key or alternate unique key. If founf replaces old values with new values.

The id = 0 is not wrong. 
If you go to the definition of id you will see its an AutoGenerate parameter

Reply:
This onConflict = REPLACE strategy is inspired from Android FHIR SDK but it seems that they use it to write clean and precise code ([ref](https://github.com/google/android-fhir/blob/76ecd8c746b17b2297241683516a100812a5dbe2/engine/src/main/java/com/google/android/fhir/db/impl/dao/ResourceDao.kt#L117-L121
))

Currently we are going to stick to this code as the same strategy is being used in other entities but we can make improvements. Check this [issue](https://github.com/google-research/CVD-paper-mobile-camera-example/issues/39)

Reply:
This sounds fine, thanks @MJ1998 for creating the [issue](https://github.com/google-research/CVD-paper-mobile-camera-example/issues/39) to track for optimisations